react/src/components/timeline-app/Application.jsx

```diff
@@ -0,0 +1,45 @@
+import MockDataService from './MockDataService';
+import React from 'react';
+import TimelineComponent from './TimelineComponent';
+
+
+class Application extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: null,
+    };
+  }
+
+
+  componentWillMount() {
+    MockDataService.fetchData(null).then((data) => {
+      this.setState({ data });
+    });
+  }
```

Comment:
also just for info, no need to change, you could make 1 liner out of that:

```js 
MockDataService.fetchData(null).then(data => this.setState({ data }));
```

Sorry if you know it already. Also I'd define the signature of fetchData like that: fetchData(entityId = null) with default param, so that you can call it like fetchData() (no need to pass null)